app/experimenter/nimbus-ui/src/components/FormBranches/index.tsx

```diff
@@ -2,47 +2,142 @@
-  } = experiment!;
+  const initialEqualRatio = useMemo(
+    () =>
+      experiment.referenceBranch !== null &&
+      experiment.treatmentBranches !== null &&
+      experiment.treatmentBranches.every(
+        (branch) =>
+          branch && branch.ratio === experiment.referenceBranch!.ratio,
+      ),
+    [experiment],
+  );
+
+  const [
+    {
+      featureConfig: experimentFeatureConfig,
+      referenceBranch,
+      treatmentBranches,
+      equalRatio,
+    },
+    dispatch,
+  ] = useReducer(formBranchesReducer, {
+    featureConfig: experiment.featureConfig,
+    referenceBranch: experiment.referenceBranch,
+    treatmentBranches: experiment.treatmentBranches,
+    equalRatio: initialEqualRatio,
+  });
+
+  const handleAddBranch = () => dispatch({ type: "addBranch" });
+
+  const handleRemoveBranch = (idx: number) => () =>
+    dispatch({ type: "removeBranch", idx });
+
+  const handleUpdateBranch = (idx: number) => (
+    value: getExperiment_experimentBySlug_treatmentBranches,
+  ) => dispatch({ type: "updateBranch", idx, value });
+
+  const handleEqualRatioChange = (ev: React.ChangeEvent<HTMLInputElement>) =>
+    dispatch({ type: "setEqualRatio", value: ev.target.checked });
+
+  // HACK: just use the first available feature config when adding
+  // The only available indicator whether to display "Add feature config" is a non-null feature config
+  const handleAddFeatureConfig = () =>
+    dispatch({ type: "setFeatureConfig", value: featureConfig![0] });
+
+  const handleRemoveFeatureConfig = () =>
+    dispatch({ type: "removeFeatureConfig" });
+
+  const handleFeatureConfigChange = (
+    value: getConfig_nimbusConfig_featureConfig | null,
+  ) => dispatch({ type: "setFeatureConfig", value });
+
+  // TODO: EXP-505 implement save button enable/disable logic
+  const isSaveDisabled = false;
+
+  const handleSaveClick = (
+    ev: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+  ) => {
+    ev.preventDefault();
+    onSave({
+      equalRatio,
+      featureConfig: experimentFeatureConfig,
+      referenceBranch,
+      treatmentBranches,
+    });
+  };
+
+  // TODO: EXP-505 implement next button enable/disable logic
+  const isNextDisabled = false;
+
+  const handleNextClick = (
+    ev: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+  ) => {
+    ev.preventDefault();
+    onNext();
+  };
+
+  const commonBranchProps = {
+    equalRatio,
+    featureConfig,
+    experimentFeatureConfig,
+    onFeatureConfigChange: handleFeatureConfigChange,
+    onAddFeatureConfig: handleAddFeatureConfig,
+    onRemoveFeatureConfig: handleRemoveFeatureConfig,
+  };
+
   return (
     <section data-testid="FormBranches" className="border-top my-3">
       <Form className="p-2">
         <Form.Row className="my-3">
           <Form.Group controlId="evenRatio">
             <Form.Check
+              data-testid="equal-ratio-checkbox"
+              onChange={handleEqualRatioChange}
+              checked={equalRatio}
```

Comment:
This will be totally dependent on whether the server-side experiment branches are all already in equal ratio when the page loads up. There is no other indicator or flag available in the data model to decide whether to check this box by default.

Reply:
Also, having a control & treatment by default might be a thing we need to rig up on the server side? Not sure what their names & slugs would be, etc

Reply:
Oh yeah this is a discussion we've had countless times.  The pattern we generally settled on in the past was to have the server spit out the default empty configuration, ie an empty control branch and an empty treatment branch, and that way the UI can just take what the server gives it and spit it out verbatim without needing to do any custom preparation logic.  We can do that here if it's a blocker?